### ISSUE TITLE
[Bug]: transform error when use two loops in one function with @babel/plugin-transform-block-scoping@7.20.9

### ISSUE THREAD

**Issue by Erucy** [babel/babel#15308]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```
const test = (data) => {
  for (let i = 0; i < data.length; i++) {
    const x = data[i];
    data.find((item) => item === x);
  }
  for (let i = 0; i < data.length; i++) {
    const x = data[i];
    data.find((item) => item === x);
  }
}
```

### Configuration file name

.babelrc

### Configuration

{
  "presets": [
    "@babel/preset-env"
  ]
}

### Current and expected behavior

current output:
```
"use strict";

var test = function test(data) {
  var _loop = function _loop() {
    var x = data[i];
    data.find(function (item) {
      return item === x;
    });
  };
  for (var i = 0; i < data.length; i++) {
    _loop();
  }
  var _loop2 = function _loop2() {
    var x = data[i];
    data.find(function (item) {
      return item === x;
    });
  };
  for (var _i = 0; _i < data.length; _i++) {
    _loop2();
  }
};
```

the second `var x = data[i];` should be `var x = data[_i];`

### Environment

  System:
    OS: macOS 12.6.2
  Binaries:
    Node: 14.19.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.16 - /usr/local/bin/npm
  npmPackages:
    @babel/cli: ^7.17.6 => 7.17.6 
    @babel/core: ^7.20.7 => 7.20.7 
    @babel/node: ^7.16.8 => 7.16.8 
    @babel/plugin-transform-block-scoping: ^7.20.9 => 7.20.9 
    @babel/preset-env: ^7.16.11 => 7.16.11

**Comment by nskazki** [comment]
@babel/plugin-transform-block-scoping@7.20.7 is the first affected version

**Comment by hanzooo** [comment]
The Same issue happened to me，with the latest version 7.20.11

**Comment by aaronhargrove-grub** [comment]
Also having this issue on 7.20.11

**Comment by liuxingbaoyu** [comment]
@hanzooo @aaronhargrove-grub Can you provide a repo to reproduce it?

**Comment by aaronhargrove-grub** [comment]
@liuxingbaoyu [This repl](https://babeljs.io/repl/#?browsers=last%202%20versions%2C%0Anot%20dead%2C%0AChrome%20%3E%3D%2045%2C%0ASafari%20%3E%3D%2011%2C%0AFirefox%20%3E%3D%2045%2C&build=&builtIns=entry&corejs=3.21&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABMOcBCBDATgCgFCKIDOUGUApgFyIDKcAtuTaRQDQGIDeiW5RADgiLlEAX2oAlPoLDCA8gCMAVuWh4AlNTqMpUEFjAAVAJ78RnDhCFRiDcgDEANmQphyAEwCSYFFnpl4JABeHmkhcgA6XncQCHIcHF4iEEcoVkQIfV4wKAA1DEcQcnVEIIA-Lg5CK1kbbmwsDGM5YEMACxgwAHMiMVKMrPIc_MLyAG4q5DgsRBxHchtO93IAD36ABjHEJdXEAB5EBqaW9s6eiPnuqDatnZWAanuSi0JXjOtEa7P-o-bWju6RAA2ncALoTN7VD4oOD9ABSNDkADkIvxsMIcL8TgDzjD1BDITUSLZGABZDD8RRKfovSGEDDUGERDDAIjuYDsOmEBSM1ARABu7gwbI5k0IogJb3mNi-gPsWAYABlsF0mFYzKUxa8kikoECiHZyZTlMzQYgAD7m2Y61L6w0UqkRBRmkJA0H4rXSz44ojyhg0fyOeZYGjqkQhWU9P30ZVYVWhuBmCLAJb4LmzFYlcqIFYRGDuUpBEIGskOk0YLXqLUwYCzACEkd9CpjKrViciCgwAC8LkMutdnlrCI3o7H42HUSAiG0cCPmwGCsGE2YPXTRFq5_7A0uJ52e_wpzPaXSS-RvL5_LAENRj1yU1gSNQAOQKOAKJ-c9OIZyPxBP64iAoIAwI4yxYE-Q5iJ-dJChQz5QHAQrGB-kH5s--YQWuq6vOuby8HoBihMkqQEqI6ScKIHocPh-hIKeTguEMHjntMl6BBMohAA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Ctypescript&prettier=false&targets=&version=7.20.12&externalPlugins=&assumptions=%7B%7D) reproduces the issue

**Comment by aaronhargrove-grub** [comment]
@liuxingbaoyu [This repl](https://babeljs.io/repl/#?browsers=last%202%20versions%2C%0Anot%20dead%2C%0AChrome%20%3E%3D%2045%2C%0ASafari%20%3E%3D%2011%2C%0AFirefox%20%3E%3D%2045%2C&build=&builtIns=entry&corejs=3.21&spec=false&loose=false&code_lz=C4TwDgpgBAyg9gWwgGTnA1gVzFAvFAbwCgooBtASwDsATCADwC4orMEAjCAJwF1mAxNACEAhlwAq4CGR4BuIgF95RAGaYqAY2AU4VKCuFiAFCSgBnYCOARm8JDEvWANKYJQuEM2F1noC5gBKnt5UvgDy7ABWEFpEAJS2iBBBwJhcVJKQhKSmGj7A5kn8ADZW1lQQNACSVAZcCFY6evgeXj4QAHQeNJgaEEZGrZjFwE5QGmkeVMAAaiLFmBBxeAB8hKakeaEFbmJcIiBhKuIAFtQA5mZQCnjjkxDTcwsQ8qSkdVBGxRAF1HT0twADLIoH8GFAADxQPYHI6nC5mDrfKjnYAnEFg-gAaixy2IbzeWwsUDRF1uMMOxzOKLMlFoDDkG0J-X0aFuACkYGEAHIdMBiXxGClw6mXDoGOBxV4E8YssxJACyIjAEUit3xMtIImYEo6IhUZhoKhcmtI7B1aA6ADcaCJDcamaQlI6oN8CqSafwuIhkGJzhAYHksrgXaQhiMyPKkEqVVE9TwoAAfROfcPASOK5WqjrsBP4GRSl1ukmisxexAwBrFb5cQNwYMlhHlhC-rj-uuQcV_Eymz70Za4Nb0DoUGh4XD4KMQGPZkQuuIuigqT4AQg9l2brfbQc67BEAC8kQ9USc8aHG57vS2_QGd3zMGYTkZ12Wr5X5jWO0tpQSFC6X8277VtwX45ge96PkYGqalONR1A02i6Mw0GmioFBcBYzAAOTsHA7BYSavaunawDYWi0DsJgFDFHQXBYeeCiEZqtrWGRcC2iABHnqO2GjvRmoKIWv5Mh4qTpO4njDMA0qMYQNx2rASSoBg2CFqYolpHoU4lGUDyVHBcD1I0ujyAoQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Ctypescript&prettier=false&targets=&version=7.20.12&externalPlugins=&assumptions=%7B%7D) reproduces the issue

**Comment by liuxingbaoyu** [comment]
![image](https://user-images.githubusercontent.com/30521560/210785359-7e5d6e84-967a-4f3d-8394-46f34119500c.png)
Thank you for your report!

**Comment by liuxingbaoyu** [comment]
https://babeljs.io/repl/#?browsers=node%204&build=&builtIns=usage&corejs=3.21&spec=false&loose=false&code_lz=C4TwDgpgBAyg9gWwgGTnA1gVzFAvFAbwCgooBtASwDsATCADwC4orMEAjCAJwF1mBDKiDI8A3EQC-4ogDNMVAMbAKcKlBloAQvy4AKHV34gA8jIAqAC2oBzAM4BKQqRKl1cLlF0AbCMCjU6ejwoAAZRf1oGKAAeKAMjU0sbWwA6HyprYAtwgIYAajzHYldXBVVbPyybYPiTcysM20pI-jEXErKqCqhbRAgAWX4wY3YAK2CCKXbXHz8DOAB3YN16R1wAPh6-weGx8RLSeYXde2mJSSJZLR1dMgBGABoAJgeAZh57IA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.20.12&externalPlugins=&assumptions=%7B%7D
Here's a smaller reproducible repl.

There are two problems here.
1. Outdated binding information causes renaming to fail.
2. `index` was incorrectly registered to `scope.globals`.
```ts
type SomeLookup = {
   [index: number]: FooBarType[];
};
```


The real fix should be 1, which seems a bit tricky to fix.
Since our `scope.rename` is actually implemented using `traverseNode`, `rename` will not work properly when it is split into two blocks.
The way I figured it out is to do the renaming before `wrapLoopBody`, but so far I haven't been able to get TDZ to continue working properly.

@nicolo-ribaudo If you plan on doing it too feel free to do it! If not I'll keep trying tomorrow.

**Comment by JLHwung** [comment]
Fixed in v7.20.14.
